test/Libraries/DynamoPythonTests/PythonEditTests.cs

```diff
@@ -226,7 +284,17 @@ public void ReturnPythonDictionary_AsDynamoDictionary()
 
             var guid = "490a8d54d0fa4782ae18c81f6eef8306";
 
-            AssertPreviewValue(guid, new Dictionary<string, int> { { "abc", 123 }, { "def", 345 } });
+            var nodeModel = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace(guid);
+            var pynode = nodeModel as PythonNode;
+
+            foreach (var pythonEngine in pythonEngineVersionsList)
+            {
+                UpdateEnginePropertyForPythonNode(pynode, pythonEngine);
```

Comment:
Is the workspace in an unsaved state at this stage due to changes to engine property?

Reply:
The workspace doesn't have any changes as we are just modifying the Engine property on the node.

Reply:
I think that means it does have unsaved changes- the engine property is a serializable value of the node right?

Reply:
But this flag ViewModel.Model.CurrentWorkspace.HasUnsavedChanges is false.

Reply:
that sounds like a bug!?

Reply:
The python node object is also marked as unmodified. Should it be marked as modified if the workspace is expected to have any changes?

Reply:
I am look more into this.

Reply:
it makes sense to me, that if the engine is changed the node should be marked as modified as this should re-execute the graph, right?

Reply:
Added the comment above to address this.